pkg/ipcache/ipcache.go

```diff
@@ -160,7 +160,8 @@ func (ipc *IPCache) InitializeFrom(entries []*models.IPListEntry) error {
 		}
 
 		var ns, pod string
-		if e.Metadata != nil && e.Metadata.Source == string(source.Kubernetes) {
+		if e.Metadata != nil &&
+			(e.Metadata.Source == string(source.Kubernetes) || e.Metadata.Source == string(source.CustomResource)) {
```

Comment:
I'm not sure about the original rationale for filtering the source here, we could probably just check `e.Metadata != nil`. But maybe @gandro or others have some idea. In particular, this version as-is would probably break in an environment where Cilium is deployed with etcd as the backend for sharing ipcache between nodes.

Reply:
The original rationale was that the source would tell you what kind of `Name` the name refers to. `Source: kubernetes`would mean that `Name` refers to a pod name (and not e.g. a docker container name).

But we never added other kinds of sources. Instead, we repurposed the `Source` field to reflect the IPCache entry source to aid with debug-ability: https://github.com/cilium/cilium/pull/15706

Thus I agree, let's remove the source check completely.

Reply:
fixed!